lib/SessionManager.js

```diff
@@ -10,22 +10,24 @@ var _ = require('underscore');
 var path = require('path');
 var touch = require('touch');
 var logger = require('../common/logger').getLogger('SessionManager');
+const sessionDuration = config.get('sessionDuration');
+const flavorDownloaderTeardownInterval = config.get('flavorDownloaderTeardownInterval');
+
 module.exports = (function(){
     var sessionFileName = "sessionTimestamp";
     // TODO: Change function erase duplications -> Gad
-    function lastTimestampSessionModification(lastModificationTime, isFromDisk, entryId) {
-        var timeToWait;
-        if (isFromDisk) {
-            logger.debug("[%s] Checking if this is a new session", entryId);
-            timeToWait = config.get('sessionDuration');
-        }
-        else {
-            logger.debug("[%s] Checking if to call 'stop' for entry", entryId);
-            timeToWait = config.get('flavorDownloaderTeardownInterval');
-        }
+    function isTimeToStopSession(lastModificationTime, entryId) {
+        logger.debug("[%s] Checking if this is a new session", entryId);
+        var currentTime = new Date();
+        logger.debug("[%s] Entry last updated at %s; time passed = %d sec; Grace period duration: %d seconds. Time to stop session.", entryId, lastModificationTime, (currentTime - lastModificationTime)/1000, flavorDownloaderTeardownInterval/1000);
+        return ((currentTime - lastModificationTime) >= flavorDownloaderTeardownInterval);
+    }
+
+    function isTimeToCreateNewSession(lastModificationTime, entryId) {
+        logger.debug("[%s] Checking if this is a new session", entryId);
         var currentTime = new Date();
-        logger.debug("[%s] Entry was last live at %s; time passed = %d sec; Grace period duration: %d seconds", entryId, lastModificationTime, (currentTime - lastModificationTime)/1000, timeToWait/1000);
-        return ((currentTime - lastModificationTime) >= timeToWait);
+        logger.debug("[%s] Entry was last live at %s; time passed = %d sec; Grace period duration: %d seconds. Need to create new live session.", entryId, lastModificationTime, (currentTime - lastModificationTime)/1000, sessionDuration/1000);
```

Comment:
log line is not inline with the function's name and functionality. 
Sentence states that we Need to create a new live session but what we are actually doing is Checking if we need to create one - which means the answer can be NO but yet we still stated that we need to...